parsers/saarland.py

```diff
@@ -0,0 +1,199 @@
+# please don't copy this key, contact felix@fefrei.de to get your own instead
+API_KEY = "A1RT5M56" + "ibsGaGNZBJQg"
+APP_VERSION = "1"
+SUPPORTED_LANGUAGES = ["de", "en"]
+
+URL_BASE = (
+    'https://mensaar.de/api/%s/%s/%s/%s/'
+    % (API_VERSION, API_KEY, APP_VERSION, SUPPORTED_LANGUAGES[0])
+)
+URL_BASE_DATA = "getBaseData"
+URL_MENU = "getMenu/"
+
+UTC_DATE_STRING = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+ROLES = {
+    's': 'student',
+    'g': 'other',
+    'm': 'employee'
+}
+
+SPECIAL_NOTICES = [
+    'su',   # geschwefelt
+    'bl',   # geschwärzt
+    'gw',   # gewachst
+    've',   # vegetarisch
+    'bio',  # biologisches Essen
+    'nla',  # laktosefrei
+    'vn'    # vegan
+]
+
+
+def get_notices(notices, sub_notices=False):
+    global base_data
+
+    notices_special = []
+    notices_allergens = []
+    notices_others = []
+    notices_others_last = []
+
+    for notice_short in notices:
+        notice = base_data['notices'][notice_short]
+        if notice['isAllergen']:
+            if sub_notices:
+                notices_allergens.append(notice['displayName'])
+            else:
+                notices_allergens.append(
+                    'enthält %s' % notice['displayName'])
+        else:
+            if notice_short == 'nsf':  # ohne Schweinefleisch
+                notices_others_last.append(notice['displayName'])
+            elif notice_short in SPECIAL_NOTICES:
+                notices_special.append(notice['displayName'])
+            else:
+                if sub_notices:
+                    notices_others.append(notice['displayName'])
+                else:
+                    notices_others.append('mit %s' % notice['displayName'])
+
+    if sub_notices:
+        if len(notices_allergens) > 0:
+            notices_allergens[0] = 'enthält %s' % notices_allergens[0]
+        if len(notices_allergens) > 1:
+            notices_allergens[-2] = (
+                '%s und %s' % (notices_allergens[-2], notices_allergens[-1]))
+            del notices_allergens[-1]
+        if len(notices_others) > 0:
+            notices_others[0] = 'mit %s' % notices_others[0]
+        if len(notices_others) > 1:
+            notices_others[-2] = (
+                '%s und %s' % (notices_others[-2], notices_others[-1]))
+            del notices_others[-1]
+
+    return (notices_special + notices_allergens +
+            notices_others + notices_others_last)
+
+
+def build_notes(notices, components):
+    global base_data
+
+    components_all = []
+
+    notices_list = get_notices(notices)
+
+    for component in components:
+        component_string = 'dazu ' + component['name']
+        if len(component['notices']) > 0:
+            component_notices_list = get_notices(component['notices'], True)
+            component_string += ' (%s)' % ', '.join(component_notices_list)
+        components_all.append(component_string)
+
+    return components_all + notices_list
+
+
+def build_hours(opening_hours):
+    if not opening_hours:
```

Comment:
NIT: This should be `if opening_hours is None`.

Reply:
Changed.